### PR TITLE
[Merged by Bors] - chore: move NormalizedGCDMonoid ℤ to reduce imports

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

Similar to #12341

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by YaelDillies** [comment]
So do you not think this could go in `Algebra.GCDMonoid.Nat`?

**Comment by Ruben-VandeVelde** [comment]
Could - sure, though it's a bit weird to make a module called `Nat` about `Int` (though the same applies to the existing file). I lean slightly towards keeping them separate, but happy to follow whatever maintainers prefer.

**Review by YaelDillies — COMMENTED** [review]
The thing is I suspect there are quite a few theorems to be proved at the intersection of the two files, and no import consideration is stopping us from merging them. So I'd lean toward having a single file, which you may call `Algebra.GCDMonoid.NatInt`, if you think that sounds better.

File: Mathlib/Algebra/GCDMonoid/Div.lean [hunk 1]
```diff
@@ -3,11 +3,11 @@ Copyright (c) 2022 Riccardo Brasca. All rights reserved.
...
 Authors: Riccardo Brasca
 -/
-import Mathlib.Algebra.GCDMonoid.Finset
 import Mathlib.Algebra.GCDMonoid.Basic
-import Mathlib.RingTheory.Int.Basic
-import Mathlib.RingTheory.Polynomial.Content
+import Mathlib.Algebra.GCDMonoid.Finset
+import Mathlib.Algebra.GCDMonoid.Int
 import Mathlib.Algebra.GCDMonoid.Nat
+import Mathlib.RingTheory.Polynomial.Content
```
Comment (YaelDillies) [inline Mathlib/Algebra/GCDMonoid/Div.lean:10]: For a followup, this import (as any `RingTheory → Algebra` import) is suspicious. Shouldn't it go the other way around?

Comment (YaelDillies) [reply Mathlib/Algebra/GCDMonoid/Div.lean:10]: I guess the
```
## TODO
Add a typeclass to state these results uniformly.
```
explains it. Do we have an idea what such a typeclass would look like? It seems all we want is `∀ a b, a ≠ 0 → a * b / a = b`, which is pretty easy to state.

Comment (YaelDillies) [reply Mathlib/Algebra/GCDMonoid/Div.lean:10]: I have now opened #12424 to define said typeclass

**Comment by Ruben-VandeVelde** [comment]
I'll let someone else tie-break on whether to merge the files or not

**Comment by YaelDillies** [comment]
Waiting for tie break here: https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.2312364.20Move.20NormalizedGCDMonoid.20Int.20to.20reduce.20imports

**Comment by adomani** [comment]
The `no_lost_declarations` script reports

---

* 0  added declarations
* 0  removed declarations
* 21  paired declarations

---

**Comment by Ruben-VandeVelde** [comment]
!bench

**Comment by Ruben-VandeVelde** [comment]
I had hoped this would cut into the longest pole, but it seems like `RingTheory.Int.Basic` no longer is in it anyway

**Comment by YaelDillies** [comment]
Doesn't mean it's useless! But yes I agree that useful interventions on the longest pole now need to be done deeper in the library (I have flagged a few contenders for improvement on Zulip a while back).

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**